File: flutter17_mobile/lib/models/order_suggestion.dart
import 'package:flutter17_mobile/models/order.dart';
import 'package:json_annotation/json_annotation.dart';

part 'order_suggestion.g.dart';

@JsonSerializable()
class OrderSuggestion {
  String? sessionId;
  Order? oldOrder;
  Order? newOrder;
 

  OrderSuggestion({
    this.sessionId,
    this.oldOrder,
    this.newOrder,   
  });

  factory OrderSuggestion.fromJson(Map<String, dynamic> json) => _$OrderSuggestionFromJson(json);

  Map<String, dynamic> toJson() => _$OrderSuggestionToJson(this);
}

File: angular17_web/src/app/models/order-suggestion/order.suggestion.ts
import {Order} from '../order/order';


export class OrderSuggestion {
  sessionId?: string;
  oldOrder!: Order;
  newOrder!: Order;

  constructor(data?: any) {
    if (!data) return;

    this.sessionId = data.sessionId;
    this.oldOrder = data.oldOrder ? new Order(data.oldOrder) : undefined as any;
    this.newOrder = data.newOrder ? new Order(data.newOrder) : undefined as any;
  }
}
